Physics3D/geometry/triangleMesh.cpp

```diff
@@ -221,10 +276,16 @@ TriangleMesh::TriangleMesh(MeshPrototype&& mesh) noexcept :
 }
 
 IteratorFactory<ShapeVertexIter> TriangleMesh::iterVertices() const {
-	return IteratorFactory<ShapeVertexIter>(ShapeVertexIter{vertices, getOffset(vertexCount)}, ShapeVertexIter{vertices + vertexCount, getOffset(vertexCount)});
+  return IteratorFactory<ShapeVertexIter>(ShapeVertexIter{vertices, 8, 0},
+					  ShapeVertexIter{&vertices[getOffset(vertexCount) * 3 -
+								    ((getOffset(vertexCount)*3-vertexCount*3)/3) - 8*2 -
+								    1 * (vertexCount%8==0)], 8, 0});
```

Comment:
I find the math a bit hard to follow. 
It might be best to add some intermediate variables to explain the math